### ISSUE TITLE
Unnecessarily complicated version compare function

### ISSUE THREAD

**Issue by codetsunami** [HotPocketDev/core#13]
https://github.com/HotPocketDev/core/blob/374424f98fb6c54a08c093a10e7b0de5bf400b35/src/util.cpp#L55

Avoid doing character comparisons in loops wherever possible. 

Do something like this instead
```
int version_compare(const string&x, const string& y) {
    istringstream ix(x), iy(y);
    while (ix.good() || iy.good()) {
        int cx = 0, cy = 0;
        ix >> cx; iy >> cy;
        if (cx > cy) return 1;
        if (cx < cy) return -1;
        ix.ignore(); iy.ignore();
    }
    return 0;
}```

**Comment by ravinsp** [comment]
Thanks. Adjusted for some malformed version components.

```
int version_compare(const string &x, const string &y)
{
    istringstream ix(x), iy(y);
    while (ix.good() && iy.good())
    {
        int cx = 0, cy = 0;
        ix >> cx; iy >> cy;

        if (!ix.good() || !iy.good())
            return -2;

        if (cx > cy) return 1;
        if (cx < cy) return -1;

        ix.ignore();
        iy.ignore();
    }

    if (!ix.good() || !iy.good())
        return -2;

    return 0;
}
```

**Comment by ravinsp** [comment]
Fixed in #15
